### PR TITLE
feat(ui): non-closeable modal support

### PR DESCRIPTION
**Pull request by sa-r-ah**

### CODE REVIEW

**Review by ceisipeik — APPROVED** [review]

File: libs/ui/src/lib/modal/modal.component.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div class="cvi-modal__dialog" role="dialog" [attr.aria-label]="title" cdkTrapFocus>
-  <button class="cvi-modal__close" (click)="closeModal()" dataAttribute="close-modal-button" aria-label="Close">
+  <button *ngIf="closeable" class="cvi-modal__close" (click)="closeModal()" dataAttribute="close-modal-button" aria-label="Close">
     <cvi-ng-icon [name]="'close'" [height]="14"></cvi-ng-icon>
   </button>
   <ng-template [ngTemplateOutlet]="templateRef" [ngTemplateOutletContext]="{ $implicit: this, title }"></ng-template>
 </div>
-<div class="cvi-modal__backdrop" dataAttribute="modal-backdrop" (click)="closeModal()"></div>
+<div class="cvi-modal__backdrop" dataAttribute="modal-backdrop" (click)="closeable && closeModal()"></div>
```
Comment (ceisipeik) [inline libs/ui/src/lib/modal/modal.component.html:7]: maybe move the `closeable` check into the closeModal() method?

**Review by ceisipeik — APPROVED** [review]

File: libs/ui/src/lib/modal/modal.component.stories.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ export const Default = {
         <h1>{{ testModalTitle }}</h1>
         <p dataAttribute="test-content">Modal content</p>
       </ng-template>
-      <cvi-ng-button [cviNgModal]="modal" modalTitle="Some title" dataAttribute="test-button">Open modal</cvi-ng-button>
+      <cvi-ng-button [cviNgModal]="modal" modalTitle="Some title" closeable="true" dataAttribute="test-button">Open modal</cvi-ng-button>
```
Comment (certainlyakey) [inline libs/ui/src/lib/modal/modal.component.stories.ts:18]: Same as in `confirmation-modal.component.stories.ts`, this attribute probably should be turned into a live Storybook control using a `closeable` arg: 
```suggestion
      <cvi-ng-button [cviNgModal]="modal" modalTitle="Some title" [closeable]="closeable" dataAttribute="test-button">Open modal</cvi-ng-button>
```

File: libs/ui/src/lib/modal/modal.component.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ export class ModalComponent implements OnDestroy {
   templateRef!: TemplateRef<any>;
 
   title!: string;
+  closeable!: boolean;
```
Comment (certainlyakey) [inline libs/ui/src/lib/modal/modal.component.ts:20]: Should the prop have a default value?

Comment (sa-r-ah) [reply libs/ui/src/lib/modal/modal.component.ts:20]: I'll put it to true so those that are using it directly wouldn't have to change anything to have same function they have right now

**Review by certainlyakey — APPROVED** [review]

**Pull request merged**